src/components/DataTable/DataTable.tsx

```diff
@@ -112,10 +105,10 @@ export function DataTable<TData, TValue>({
                     </DropdownMenuTrigger>
                     <DropdownMenuContent align="end">
                       <DropdownMenuLabel>Actions</DropdownMenuLabel>
-                      <ViewDialog row={row} />
-                      {EditDialog ? <EditDialog row={row} /> : null}
-                      {DeleteDialog ? <DeleteDialog row={row} /> : null}
-                      {RevokeKeyDialog ? <RevokeKeyDialog row={row} /> : null}
+                      <ViewDialog row={row as Row<RowType>} />
+                      {renderItem ? renderItem(row as Row<RowType>) : null}
```

Comment:
This is better 
but still imho there should be 2 functions:
`renderActions` which can handle that the row is contact and what actions can be applied to it
and `renderInlineActions`